### ISSUE TITLE
Optional environment variable using null value

### ISSUE THREAD

**Issue by feliskio** [vercel/terraform-provider-vercel#95]
I'd like to optionally declare an environment variable within `vercel_deployment` using a ternary/conditional expression by either setting a string or null.
For example:

```
environment = {
    MY_VAR = terraform.workspace == "default" ? "value" : null
}
```

This however throws an `unhandled null value` exception.

The only workaround right now is to instead set an empty string:

```
environment = {
    MY_VAR = terraform.workspace == "default" ? "value" : ""
}
```

I believe it would be a good addition also allow null values, and just ignore the `environment` entry then.

**Comment by dglsparsons** [comment]
Hey @feliskio. Thanks for raising this. I like the idea, but I'm a bit unclear what `resource` you are using with the environment variables here. Is this for a `vercel_project`, or maybe a `vercel_deployment`? 

There's a dedicated `vercel_project_environment_variable` resource ([docs])(https://registry.terraform.io/providers/vercel/vercel/latest/docs/resources/project_environment_variable). 

I think you could use this with a `count` argument to omit setting the environment variable. e.g. 

```terraform
resource "vercel_project_environment_variable" "example" {
  count      = terraform.workspace == "default" ? 0 : 1
  project_id = vercel_project.example.id
  key        = "foo"
  value      = "bar"
  target     = ["production"]
}
```

Not sure if that solves your issue though? :)

**Comment by feliskio** [comment]
Sorry about that oversight! I was referring to the `vercel_deployment` resource and defining environment variables on a per-deployment basis.

**Comment by dglsparsons** [comment]
Makes sense! I like the suggestion. I guess it may be possible at the moment using `merge`, but may look a bit messy if you have a lot of variables with different conditions. 
(https://developer.hashicorp.com/terraform/language/functions/merge).  

e.g.

```terraform
resource "vercel_deployment" "foo" {
   project_id = vercel_project.foo.id
   
   environment = merge(
     {
        always = "present", 
        env = "variables",
     }, 
     terraform.workspace == "default" ? {} : { my_var = "value" }, 
   )
}

**Comment by dglsparsons** [comment]
Hi, this is resolved in `v0.11.4`. Please give it a go and let me know if you find anything else! 🙌

**Comment by feliskio** [comment]
Ran without any problems - thank you! 👌
